### PR TITLE
Migrated Software Docs From GitBook To Markdown Files

### PR DESCRIPTION
**Pull request by garethellis0**
### Description
Migrated all software-specific documentation from GitBook to a system of markdown files.

This will allow tight coupling between adding code and adding documentation.

### Testing Done
Checked that links between the markdown files work.

### Length Justification

Docs have lots of lines :stuck_out_tongue_winking_eye: 

### Review Checklist

**_It is the reviewers responsibility to also make sure every item here has been covered_**

- [x] **Start of document comments**: each `.cpp` and `.h` file should have a comment at the start of it. See files in the `thunderbots/software/geom` folder for examples.
- [x] **Function comments**: All function definitions (usually in the `.h` file) should have a javadoc style comment at the start of them. For examples, see the classes defined in `thunderbots/software/geom`
- [x] **Remove all commented out code**
- [x] **Remove extra print statements**: for example, those just used for testing
- [x] **Resolve all TODO's**: All `TODO` (or similar) statements should either be completed or associated with a github issue
- [x] **Justify drops in code coverage**: If this PR results in a non-trivial drop in code coverage (a bot should post a coverage diagram as a comment), please justify why we can't test the code that's not covered.

### CODE REVIEW

File: docs/code-style-guide.md [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+  float distance = catch_distance * SCALE_FACTOR;
+  ```
+
+### Comments
+
+Make sure to comment both the interface for a function or class, as well as the logic in the implementation. In general, try to make as many in-code documentations whenever possible.
+
+As much code documentation as possible should live with the code itself \(in the form of comments\). This makes it easier for people working on the code to find the information, and because the code and comments are version-controlled together if we ever go back to an older version of the code, we will have the corresponding older documentation as well.
+
+*Code comments are very important. Be sure to comment your code well enough so that another member of the team would be able to quickly get an understanding of what your code is doing, **and why**. Try not to make your comments too unnecessarily verbose, but include as much detail you feel is necessary to get the necessary points across. We realize that sounds contradictory, but use your best judgement as to what you think is clear and readable.*
```
Comment (EvanMorcom) [inline docs/code-style-guide.md]: UBER nit warning:

I would remove the 'too'. Just saying 'unessisary' would flow a lot better.

Comment (EvanMorcom) [inline docs/code-style-guide.md]: UBER nit warning 2:

There are a lot of 'neccisarries' going on here.

I would rephrase to 'but include as much detail you feel is required to adequately explain the code'

**Review by EvanMorcom — COMMENTED** [review]
Couple Nits.

Can we talk about sharing some of the TODO's?

File: docs/code-style-guide.md (hunk 1 above)
Comment (garethellis0) [reply docs/code-style-guide.md]: Hmm, removed `too`, but left the rest as-is. Let me know if you feel strongly about this though and I can reconsider.

Comment (garethellis0) [reply docs/code-style-guide.md]: changed

**Comment by garethellis0** [comment]
Comments addressed. What exactly do you mean by "sharing the TODO's"?

**Comment by EvanMorcom** [comment]
@garethellis0  I just mean create separate issues for all the TODOs so other people can pick up some of the work.

**Comment by garethellis0** [comment]
Ah, this is in a (currently offline) large list of things to make issues for. Is it cool if I push off doing this until all them get converted to issues (sometime this week hopefully)?

**Comment by EvanMorcom** [comment]
👍

**Review by MathewMacDougall — APPROVED** [review]
